### DISCUSSION TITLE
Does anyone use -t ?

### DISCUSSION THREAD

**Discussion by bongobongoland** [Pennyw0rth/NetExec#31] [Q&A] [answered]
I'm trying to figure out why it's not working.

**Comment by Marshall-Hallenbeck** [comment]
How do you know its not working? By default it's already pretty high @ 100. That argument is passed directly into the ThreadPoolExecutor https://github.com/Pennyw0rth/NetExec/blob/main/nxc/netexec.py#L58 & line 63

**Comment by bongobongoland** [comment]
Hmm... Interesting. I've tried setting different -t and did not observe any difference. Still goes thru the hosts slowly, one at a time. when using smb

**Reply by Marshall-Hallenbeck** [reply]
What version are you on, and how do you know it's only doing 1 at a time? How many total hosts are you scanning at once?

**Reply by bongobongoland** [reply]
My understanding is that the idea behind multithreading implies connecting to multiple hosts at once. So, if I have, say, 30 hosts with -t 30, we'll be connecting to 30 hosts at the same time. smb hosts.txt -u admin -p P@ss --local-auth will spit out many results rather quickly. I'm using the latest version.

**Reply by Marshall-Hallenbeck** [reply, marked as answer]
We create a ThreadPooler task for each thread that is defined (by default it's 100, which is plenty): https://github.com/Pennyw0rth/NetExec/blob/main/nxc/netexec.py#L55

There are default timeouts and network connection timing that we can't help - what specifically makes you think that the threads aren't being properly created? If it wasn't multi-threaded, you would see each host 1 by 1, but you see results randomly (that is, sometimes you see them slightly out of order).
